src/collector.ts

```diff
@@ -126,21 +126,41 @@ class NaiveGomodParser {
                         entry.value = new Variant(ValueType.String, 'v' + version[0]);
                         entry.value_position = { line: index + 1, column: version.index };
                         dependencies.push(new Dependency(entry));
-
-                        // Find packages of this module.
-                        goImports.forEach(pckg => {
-                            if (pckg != pkgName && pckg.startsWith(pkgName + "/")) {
-                                const entry: IKeyValueEntry = new KeyValueEntry(pckg, { line: 0, column: 0 });
-                                entry.value = new Variant(ValueType.String, 'v' + version[0]);
-                                entry.value_position = { line: index + 1, column: version.index };
-                                dependencies.push(new Dependency(entry));
-                            }
-                        })
                     }
                 }
             }
             return dependencies;
         }, []);
+
+        let dependencies = []
+        goImports.forEach(importStatement => {
+            let exactMatchDep: Dependency = null;
+            let moduleMatchDep: Dependency = null;
+            goModDeps.forEach(goModDep => {
+                if (importStatement == goModDep.name.value) {
+                    // Stack uses module directly
+                    exactMatchDep = goModDep;
+                    dependencies.push(goModDep);
```

Comment:
Just for the sake of code organization, can this `dependencies.push(goModDep);` be pushed below.?

Reply:
This is done. PTAL.